patroni/ha.py

```diff
@@ -1279,7 +1285,7 @@ def _run_cycle(self):
             # is data directory empty?
             if self.state_handler.data_directory_empty():
                 self.state_handler.set_role('uninitialized')
-                self.state_handler.stop('immediate')
+                self.state_handler.stop('immediate', stop_timeout=self.master_stop_timeout())
```

Comment:
If the data directory isn't present or empty we want to make sure that postgres isn't running. It would theoretically help with the case when postgres is alive, but doesn't respond to immediate shutdown.

Reply:
Agree, especially in case where the data dir is empty and the master is hung. Should we give precedence to the stop timeout set in config and override it only if its null?

Reply:
The same as https://github.com/zalando/patroni/pull/1445/files#diff-57c53e988b775ee2902488fb33c1f433R1092
```suggestion
                self.state_handler.stop('immediate', stop_timeout=self.patroni.config['retry_timeout'])
```